docs/docs/canonicai/chapter3.md

```diff
@@ -0,0 +1,339 @@
+    }
+}
+```
+
+### confirmed state
+This state is triggered after the user have positively agreed that all the data provided is correct and based on the information provided the company will makes it decision afterwards. In this case the sales person will tell the user "You are all set for a tesla test drive". Below shows how the node is created. 
+
+```
+node confirmed:cai_state {
+    has name = "confirmed";
+    can gen_response {
+        visitor.response = "You are all set for a Tesla test drive!";
+    }
+}
+```
+
+### canceled state
+This state is triggered when the user wants to hop out of the entire conversation and it's done through intent classification. 
+
+```
+node canceled:cai_state {
+    has name = "canceled";
+    can gen_response {
+        visitor.response = "No worries. We look forward to hear from you in the future!";
+    }
+}
+```
+
+### **Edges**
+### intent_transition edge
+This edge allows you to transition from state to state (node to node) when provided an intent which is passed as a parameter. For e.g. when the user ask "Can I test drive the tesla" it will go through the bi-encoder AI model which is used for intent classification and find the intent and through the edge it will pass the intent to the intent_transition edge which will search for which node to travel to, to move to the next state. In this case it will be the collect_info state.
+
+```
+edge intent_transition {
+    has intent;
+}
+```
+
+### entity_transition edge
+This edge allows you to transition from state to state (node to node) when provided all the entities needed which is passed as a parameter. For e.g. when the user have to provide information in the collect_info state "My name is Jemmott I live in lot 1 pineapple street" it will go through the entity extraction AI model, extract "Jemmott" as the name and "lot 1 pineapple street" as the address and it will passed these information into the entity_transition edge and move you on to the next state, in this case it will be the confirmed state. If all the information is not provided it will not transition to the next state until all the entities are fulfilled. 
+
+```
+edge entity_transition {
+    has entities;
+} 
+```
+
+### **Graph Definition**
+### spawning nodes
+Nodes are the states we used to build out the conversation AI experiences. Below, shows how to create states (nodes) using jac.
+```
+spawn {
+    state_cai_root = spawn node::cai_root;
+    state_collect_info  = spawn node::collect_info;
+    state_confirmation = spawn node::confirmation;
+    state_confirmed = spawn node::confirmed;
+    state_canceled = spawn node::canceled;
+}
+```
+
+### connecting the nodes with the two types of edges
+There are two types of edges: intent_transition edge and entity_transition edge and we will use these to connect states (nodes) to each other. 
+
```

Comment:
Let's use two statements from the code below as examples to illustrates how intent and entity transition can be leveraged. You introduced intent/entity transitions earlier but there are no concrete examples yet.

Reply:
I'm kinda confuse here too like in addition to the code snippet use an example to plug into the code statment one for intent classification and one for entity extraction.

Reply:
What I meant was, breaking down the large spawn block below into smaller sections and explain how the intent and entity transition works in those examples. 

For instance, something like this,

```
        state_collect_info -[entity_transition(
            entities = ["name", "address"]
        )]-> state_confirmation;
```
"Here we have an entity transiton going from the collect_info state to the confirm state. This entity transition has its entities variable set as "name" and "address", which will be checked later in the talk walker to evaluate if this transition should be triggered or not."